packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -92,3 +93,27 @@ export function renderMeshOwner(resourceAccount: string, meshAccount: string) :
     ? meshAccount
     : undefined;
 }
+
+/**
+ * This is the helper method to check if properties for passed match are undefined.
+ */
+export function isMatchPropertiesUndefined(match: HttpRouteMatch | undefined): boolean {
+  let isEmpty = true;
+  for (const property in match) {
+    if (property) {
+      isEmpty = false;
+      break;
+    }
+  }
+
+  return isEmpty;
+}
+
+/**
+ * This is the helper method to validate the length of Metadata array when it is specified.
+ */
+export function validateMetadata(metadata?: GrpcMetadataMatch[]) {
+  if (metadata && (metadata.length < 1 || metadata.length > 10)) {
+    throw new Error('Metadata must be between 1 and 10');
```

Comment:
Please include how the value that caused the error in the message.

Reply:
Sorry for using the magic number. Will add those.